proposals/0000-rich-errors-proposal.rst

```diff
@@ -255,6 +304,33 @@ be useful. We hope that the discussion that arises from this proposal will shed
 light on additional items. Moreover, we anticipate that the vocabulary will
 grow in time as new tooling applications are found.
 
+A smaller but very concrete challenge is figuring out how to give users a hook
+into the processing of error documents when they're reported (and possibly
+in other places where our prototype implementation had to "strip off"
+annotations) because of the lack of such a hook. Our prototype just applies
+the simplest ``SDoc' a -> SDoc`` function, while a user supplied function of
+type ``a -> SDoc`` for a suitable annotation type would let GHC adapt the
+final document, depending on the needs. One close solution is the ``log_action``
+field in ``DynFlags``, but it currently takes an ``SDoc``, and is probably not
+the only "document consumer" that would have to be updated. Any specific choice
+of annotation type would make it useless for "clients" that need another one
+(or none).
+
+Variations
```

Comment:
I would move these two the "alternatives" section to make it clear that these are not the designs we are proposing.

Reply:
Done in upcoming revision.